lua/core/softcut.lua

```diff
@@ -333,17 +333,21 @@ end
 -- @tparam number dur : duration in seconds. if -1, read as much as possible.
 -- @tparam int ch_src : soundfie channel to read
 -- @tparam int ch_dst : buffer channel to write
-SC.buffer_read_mono = function(file, start_src, start_dst, dur, ch_src, ch_dst)
-  _norns.cut_buffer_read_mono(file, start_src, start_dst, dur, ch_src, ch_dst)
+-- @tparam number preserve : level of existing material
+-- @tparam number mix : level of new material
+SC.buffer_read_mono = function(file, start_src, start_dst, dur, ch_src, ch_dst, preserve, mix)
+  _norns.cut_buffer_read_mono(file, start_src, start_dst, dur, ch_src, ch_dst, preserve or 0, mix or 1)
```

Comment:
ooh i like putting the fallback here, very nice

Reply:
we should definitely add more defaults, don't know why we haven't before

Reply:
we could even go so far as to say 

```lua
cut_buffer_read_mono(file, start_src or 0, start_dst or 0, 
dur or -1, ch_src or 1, ch_dst or 1, preserve or 0, mix or 1)
```

Reply:
Yeah exactly

Reply:
I'm happy to add those to `softcut.lua` if that's not outside the scope of this PR!

Reply:
just added! couldn't think of a sensible default for `samples` in `buffer_render`, so I left that one as-is.